### PR TITLE
Fix forecasting reduction test for regressors

### LINKED ISSUE #64: Forecasting reduction capabilities/tests
Had a quick look into the forecasting wrapper problem from sktime 0.4, brought up in PR #59, referenced now in #47.

Ultimately the problem is that the reduction strategy passes the data to the -dl regressor as numpy arrays (even if pd.Series are given to the ReducedRegressionForecaster.fit()), which gets flagged as illegal by sktime.utils.validation.series_as_features.check_X_y, which is always checked because input_checks=True is the default in the -dl regressor's fit() 

The test ([currently commented out on dev](https://github.com/sktime/sktime-dl/blob/dev/sktime_dl/deeplearning/tests/test_regressors.py#L40)) basically follows the code the the forecasting example from base sktime, which uses the KNNRegressor from sklearn, which  of course doesn't run with the check_X_y

This comes down to what is the convention now in base sktime in terms of data input/throughput. Looking at a couple of the classifiers in base sktime, [e.g.](https://github.com/alan-turing-institute/sktime/blob/master/sktime/classification/interval_based/_tsf.py#L117), the checks are applied unconditionally, so presumably regression reduced to classification runs into the same problem? 

1. We relax the dataframe requirement (which technically under the current code is only a requirement if input_checks=True, which again it is by default) and allow np arrays
2. We change the default to input_checks=False
3. This is a problem in the reduction strategy in that it should be passing around dataframes
4. This is a problem in the reduction strategy in that it should try to pass the keyword arg input_checks=False to the base regressor's fit, under the assumption that it produces data (via the splitter etc.) that is suitable anyway and doesn't need checking, [e.g. here ](https://github.com/alan-turing-institute/sktime/blob/master/sktime/forecasting/compose/_reduce.py#L307)
5. ?? 

Thoughts?

### PR DESCRIPTION
**Pull request by mloning**
#### Reference Issues/PRs
Fixes #64

### CODE REVIEW

**Comment by James-Large** [comment]
Ah almost went of wild goose chase, I commited directly onto the PR a change reflecting that the model lists are dicts now instead of lists.

**Comment by James-Large** [comment]
So something's going on with this on tensorflow version 1.15 specifically, all other versions are passing. 

Most (but not all) of the predictions being produced (in y_pred, which is being passed to mean_squared_error()) are NaN. The point of failure has been MCDCNNRegressor across azure/travis, but maybe not the only one. 

MCDCNNRegressor as a preprocessing step, prepare_input(), which may be messing it up (data to small, etc). I'll look into it this afternoon/evening.

Meantime I'll make a commit here just removing MCDCNNRegressor from the model lists and confirm everything else passes

**Comment by James-Large** [comment]
Well, MCDCNNRegressor now gets double the epochs to work with during the forecasting test. On my tf1.15 python 3.6 env I could recreate the error about half the time, so I _think_ it's just an error of nothing being learned, exploding gradients, or whatever else based on the initialisation. 

Why that would happen only on 1.15 is a mystery, though that it happened there only could have been coincidence. Just another day in deep learning land

If we're willing to call that solved, this is all good to merge I think
